### DISCUSSION TITLE
Authentication failed

### DISCUSSION THREAD

**Discussion by AndyK28** [jstedfast/MailKit#1830] [Q&A]
Hi, I am using MailKit for the first time and I want to get emails from my gmail account.

This is the error I am encountering:
Unhandled exception. MailKit.Security.AuthenticationException: Authentication failed.
   at MailKit.Net.Imap.ImapClient.ProcessAuthenticateResponse(ImapCommand ic, SaslMechanism mechanism)
   at MailKit.Net.Imap.ImapClient.Authenticate(SaslMechanism mechanism, CancellationToken cancellationToken)
   at PasswordRetrievalService.FetchEmail(String username, String password, String senderEmail)

This is my code so far:
```c#
public void FetchEmail(string username, string password, string senderEmail)
    {
        using (var client = new ImapClient(new ProtocolLogger ("imap.log")))
        {
            client.Connect("imap.gmail.com", 993, SecureSocketOptions.SslOnConnect);
            client.Authenticate(username, password);

            var inbox = client.Inbox;
            inbox.Open(FolderAccess.ReadOnly);

            Console.WriteLine("Total messages: {0}", inbox.Count);
            Console.WriteLine("Recent messages: {0}", inbox.Recent);

            var query = SearchQuery.FromContains(senderEmail).And(SearchQuery.BodyContains("password")).And(SearchQuery.NotSeen);
            var results = inbox.Search(query);

            if (results.Count == 0)
            {
                Console.WriteLine("No emails found from the specified sender containing the word 'password'.");
            }
            else
            {
                foreach (var uniqueId in results)
                {
                    var message = inbox.GetMessage(uniqueId);
                    Console.WriteLine("Subject: {0}", message.Subject);
                    Console.WriteLine("Body: {0}", message.TextBody);  // For plain text bodies
                }
            }

            client.Disconnect(true);
        }
    }
```

I do not have 2FA for this account and I do not want to add one because I am using it as part of test automation. Please advice on how to move forward 🙏🏾

**Comment by jstedfast** [comment]
I've answered this question in the [FAQ](https://github.com/jstedfast/MailKit/blob/master/FAQ.md#gmail-access).

You either need to enable 2FA and create an "app password" -or- you need to follow the directions for authenticating with OAuth2 which I've documented here: [GMailOAuth2.md](https://github.com/jstedfast/MailKit/blob/master/GMailOAuth2.md)

Google does not support any other way. Sorry :(
